includes/core/server/request/class-list-transactions.php

```diff
@@ -0,0 +1,279 @@
+			'date_between'             => $date_between_filter,
+			'type_is'                  => $request->get_param( 'type_is' ),
+			'type_is_not'              => $request->get_param( 'type_is_not' ),
+			'store_currency_is'        => $request->get_param( 'store_currency_is' ),
+			'customer_currency_is'     => $request->get_param( 'customer_currency_is' ),
+			'customer_currency_is_not' => $request->get_param( 'customer_currency_is_not' ),
+			'loan_id_is'               => $request->get_param( 'loan_id_is' ),
+			'search'                   => $request->get_param( 'search' ),
+		];
+
+		$wcpay_request->set_filters( $filters );
+		return $wcpay_request;
+	}
+
+	/**
+	 * Set match.
+	 *
+	 * @param string $match Match.
+	 *
+	 * @return void
+	 */
+	public function set_match( string $match ) {
+		$this->set_param( 'match', $match );
+	}
+
+	/**
+	 * Set loan id is.
+	 *
+	 * @param string $loan_id Loan id.
+	 *
+	 * @return void
+	 */
+	public function set_loan_id_is( string $loan_id ) {
+		$this->set_param( 'loan_id_is', $loan_id );
+	}
+
+	/**
+	 * Set currency is.
+	 *
+	 * @param string $currency Store currency.
+	 *
+	 * @return void
+	 */
+	public function set_store_currency_is( string $currency ) {
+		$this->set_param( 'store_currency_is', $currency );
+	}
+
+	/**
+	 * Set customer currency is.
+	 *
+	 * @param string $customer_currency_is Store currency.
+	 *
+	 * @return void
+	 */
+	public function set_customer_currency_is( string $customer_currency_is ) {
+		$this->set_param( 'customer_currency_is', $customer_currency_is );
+	}
+
+	/**
+	 * Set customer currency is not.
+	 *
+	 * @param string $currency Store currency.
+	 *
+	 * @return void
+	 */
+	public function set_customer_currency_is_not( string $currency ) {
+		$this->set_param( 'customer_currency_is_not', $currency );
+	}
+
+	/**
+	 * Set search.
+	 *
+	 * @param string $search Search term.
+	 *
+	 * @return void
+	 */
+	public function set_search( string $search ) {
+		$this->set_param( 'search', $search );
+	}
+
+	/**
+	 * Set date after.
+	 *
+	 * @param string $date_after Date after.
+	 *
+	 * @return void
+	 * @throws \WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception
+	 */
+	public function set_date_after( string $date_after ) {
+		$this->validate_date( $date_after );
+		$this->set_param( 'date_after', $date_after );
+	}
+
+	/**
+	 * Set date before.
+	 *
+	 * @param string $date_before Date before.
+	 *
+	 * @return void
+	 * @throws \WCPay\Core\Exceptions\Server\Request\Invalid_Request_Parameter_Exception
+	 */
+	public function set_date_before( string $date_before ) {
+		$this->validate_date( $date_before );
+		$this->set_param( 'date_before', $date_before );
+	}
+
+	/**
+	 * Set date between.
+	 *
+	 * @param array $date_between Date between.
+	 *
+	 * @return void
+	 */
+	public function set_date_between( array $date_between ) {
+		// Add only if it's not empty.
+		if ( $date_between ) {
```

Comment:
Since we're validating dates in the other methods, let's do it here as well. Considering the parameter is an array with numeric indexes (0 and 1), how about something like this?

```suggestion
		if ( is_array( $date_between ) && ! empty( $date_between ) ) {
			foreach ( $date_between as $date ) {
				$this->validate_date( $date );
			}
			$this->set_param( 'date_between', $date_between );
		}
```

Reply:
This is refactored and moved inti date params trait.